flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/triggers/DeferredComputationTrigger.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.table.runtime.triggers
+
+import org.apache.flink.streaming.api.windowing.triggers.{Trigger, TriggerResult}
+import org.apache.flink.streaming.api.windowing.triggers.Trigger.{OnMergeContext, TriggerContext}
+import org.apache.flink.streaming.api.windowing.windows.TimeWindow
+import org.apache.flink.table.api.StreamQueryConfig
+
+/**
+  * A [[Trigger]] that onElement method never fires the window, and only fires when current
+  * watermark not smaller than (window.maxTimestamp + queryConfig.getDeferredComputationTime)
+  */
+class DeferredComputationTrigger(queryConfig: StreamQueryConfig) extends Trigger[Any, TimeWindow] {
```

Comment:
I think the parameter can be only a `deferredComputationTime : Long`. It shouldn't be associated to StreamQueryConfig.